Meshtastic/Views/Nodes/EnvironmentMetricsLog.swift

```diff
@@ -219,4 +243,49 @@ struct EnvironmentMetricsLog: View {
 			}
 		)
 	}
+
+	// Set and optionally adjust the chart min/max
+	func updateChartRange(chartLowTemp: Double, chartHighTemp: Double, toggleZoom: Bool) {
+		if toggleZoom {
+			chartZoomed.toggle()
+		}
+		if chartZoomed {
+			// Use the low and high temp for the bounds
+			chartMin = chartLowTemp - Double(5.0)
+			chartMax = chartHighTemp + Double(5.0)
+		} else {
+			// Use the default low and high bounds, adjust if the high or low land outside the
+			// default min/max
+			chartMin = (chartLowTemp < chartDefaultMin ? chartLowTemp : chartDefaultMin) - Double(5.0)
+			chartMax = (chartHighTemp > chartDefaultMax ? chartHighTemp : chartDefaultMax) + Double(5.0)
+		}
+		return
+	}
+}
+
+// Set up gradient stops relative to the scale of the temperature chart
```

Comment:
This is currently implemented for mapping temperature to gradient stops. But this could be used for other gradient mapping by modifying the stopTargets array elements.